File: UtopianTree.php
<?php

// Complete the utopianTree function below.
function utopianTree($n) {
    $h=1;
    if ($n==0) {
        return($h); 
    }else {
        for ($i=1; $i <= $n; $i++) { 
            if ($i%2!=0) {
                $h=$h*2;
            }else {
                $h=$h+1;
            }
        }
    }
    return($h);


}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$stdin = fopen("php://stdin", "r");

fscanf($stdin, "%d\n", $t);

for ($t_itr = 0; $t_itr < $t; $t_itr++) {
    fscanf($stdin, "%d\n", $n);

    $result = utopianTree($n);

    fwrite($fptr, $result . "\n");
}

fclose($stdin);
fclose($fptr);

File: UtopianTree.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the utopianTree function below.
def utopianTree(n):
    h=0
    if n==0:
      return 1
    else:
        h=1
        for i in range(n):
            if i%2!=0:
                h=h+1
            else:
                h=h*2
    return h
           
           

if __name__ == '__main__':
    

    n=4
    result = utopianTree(n)

        
